src/app/shared/comment-editor/upload-text-insertor.ts

```diff
@@ -0,0 +1,62 @@
+import { ElementRef } from '@angular/core';
+import { AbstractControl } from '@angular/forms';
+
+const DISPLAYABLE_CONTENT = ['gif', 'jpeg', 'jpg', 'png'];
+
+export function insertUploadingText(
+  filename: string,
+  commentField: AbstractControl,
+  commentTextArea: ElementRef<HTMLTextAreaElement>
+): string {
+  const originalDescription = commentField.value;
+
+  const fileType = filename.split('.').pop();
+  let toInsert: string;
+  if (DISPLAYABLE_CONTENT.includes(fileType.toLowerCase())) {
+    toInsert = `![Uploading ${filename}...]\n`;
```

Comment:
I think it might be better to extract out all these repeated `[Uploading ${filename}...]` strings into a static variable.